### PR TITLE
ci: adding path ignore to documentations

### LINKED ISSUE #12287: Integration tests should not run if only documentation is updated
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
All github actions are currently triggered by default for all PRs. If only documentation is updated, there is no need to run the integration tests. 

**Expected behavior:**
We can use [paths-ignore](https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#using-filters-to-target-specific-paths-for-pull-request-or-push-events) to skip the integration test actions if only the Documentation path is updated. For example, this trigger could be added in the Canary tests, and all the integration-test-*.yaml files (wherever the `skip-ci` label is currently used):

```yaml
on:
  pull_request:
    paths-ignore:
      - 'Documentation/**'
      - 'design/**'
```

This will minimize our need for the `skip-ci` label on the PRs.

### PR DESCRIPTION
**Pull request by Mourishitz**
**Description of your changes:**
Updating GitHub Actions Workflows so it ignores the documentations changes.

**Which issue is resolved by this Pull Request:**
Resolves #12287 

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [x] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by subhamkrai — APPROVED** [review]

File: .github/workflows/canary-integration-test.yml
Comment (Mourishitz) [reply .github/workflows/canary-integration-test.yml:14]: Sure, I will add some documentation change and we can see it working.

The double start at the end was suggested in the issue description, so I tought it would be better to keep it

Comment (parth-gr) [reply .github/workflows/canary-integration-test.yml:14]: > The double star

Can you look why it is needed,

Good for your understanding

Comment (parth-gr) [reply .github/workflows/canary-integration-test.yml:14]: Yaa thats right,

I guess testing by adding a second commit won't help, as the current CI would be triggered what we have already in the code,

So you can remove the test commit

**Review by parth-gr — APPROVED** [review]

File: .github/workflows/canary-integration-test.yml
Comment (Mourishitz) [reply .github/workflows/canary-integration-test.yml:14]: I just removed it on: [3c02f40](https://github.com/rook/rook/pull/12291/commits/3c02f402b3c6664eb3769b25eba6597d3558dad9)

File: .github/workflows/integration-tests-on-release.yaml
Comment (travisn) [inline .github/workflows/integration-tests-on-release.yaml:9]: Let's actually revert this one. While it's not strictly necessary, it's nice to have the release tests run more often, even if not affected by the docs.

**Review by travisn — CHANGES_REQUESTED** [review]
Please squash the commits to a single commit, thanks!

File: .github/workflows/integration-tests-on-release.yaml
Comment (travisn) [reply .github/workflows/integration-tests-on-release.yaml]: It's still showing three commits, did you force push the new commit to the branch?

Comment (travisn) [reply .github/workflows/integration-tests-on-release.yaml]: Actually, it looks like you picked up a couple extra commits from the history. Try `git rebase upstream/master`, then force push again

Comment (Mourishitz) [reply .github/workflows/integration-tests-on-release.yaml]: Yes, it is even on the GitHub Actions run, I don´t know why is it showing commits [abb5a12](https://github.com/rook/rook/pull/12291/commits/abb5a12afd9eeafc00d273b75115abecf9ce44a2) and [94b533c](https://github.com/rook/rook/pull/12291/commits/94b533cb3ba0d970b89b87696cad21e4fba5842c) since my git rebase command only got my own commits. For some reason they show up as a part of the pull request

image

Comment (Mourishitz) [reply .github/workflows/integration-tests-on-release.yaml]: If you click on [1222fe7](https://github.com/rook/rook/pull/12291/commits/1222fe7275647ebb3bdea989fab37ea5143ef173) it even has the combination of my 3 commits message's in there (confirming the successfull squash)

Comment (travisn) [reply .github/workflows/integration-tests-on-release.yaml]: Did you see [this comment](https://github.com/rook/rook/pull/12291#discussion_r1210576993)?

Comment (Mourishitz) [reply .github/workflows/integration-tests-on-release.yaml]: Now it should be good to go! I'm sorry for some Git issues I've commited on this run, still not in an expert level of the tool..

Hope this solves the issue! Thanks for letting me contribute to the project!

File: .github/workflows/integration-tests-on-release.yaml
Comment (travisn) [inline .github/workflows/integration-tests-on-release.yaml:9]: Please revert this change, no need for it now

**Review by travisn — CHANGES_REQUESTED** [review]
The single commit looks good, now please just edit the commit description so there aren't three separate messages, thanks.

File: .github/workflows/integration-tests-on-release.yaml
Comment (Mourishitz) [reply .github/workflows/integration-tests-on-release.yaml:9]: This means to pur back the 'paths-ignore' or simply just remove the blank line? I will use this to ammend and push with a better commit message

Comment (travisn) [reply .github/workflows/integration-tests-on-release.yaml:9]: I mean to add back that blank line so no changes show for this file.

**Review by travisn — CHANGES_REQUESTED** [review]
Could you still add a brief description in the commit message, instead of just the commit title? See [this topic](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure). Thanks for responding to all the small requests. :)

**Comment by Mourishitz** [comment]
I've just read the [development flow guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure) and re-wrote my commit description, hope it fits on the repository pattern :)
> c29765e

**Comment by travisn** [comment]
> > Could you still add a brief description in the commit message, instead of just the commit title? See [this topic](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure). Thanks for responding to all the small requests. :)
> 
> I've just read the [development flow guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure) and re-wrote my commit description, hope it fits on the repository pattern :)
> 
> > [c29765e](https://github.com/rook/rook/commit/c29765e55fee0846315f1cc69e90357634f4b84a)

Perfect thanks, now I'll just wait for the CI to finish before merging...

**Review by travisn — APPROVED** [review]

**Pull request merged**
